src/main/java/io/cdap/plugin/ariba/source/AribaBatchSource.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.ariba.source;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.batch.BatchSourceContext;
+import io.cdap.plugin.ariba.source.config.AribaPluginConfig;
+import io.cdap.plugin.ariba.source.exception.AribaException;
+import io.cdap.plugin.ariba.source.util.AribaUtil;
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import org.apache.hadoop.io.NullWritable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+
+/**
+ * A {@link BatchSource} that reads data from Ariba
+ */
+
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name(AribaBatchSource.NAME)
+@Description("Reads data from Sap Ariba. " +
+  "Outputs one record for each row in each table, with the table name as a record field. " +
```

Comment:
Is this a multi source plugin? Where are we setting "pipeline argument for each table read". Also, where are we implementing this - "table name as a record field"

Reply:
Nope, it's not a multi source plugin. I had just taken those lines from ServiceNow Plugin which was pointed above.
Removed those extra lines.

Reply:
Not getting option to reply on this Query so replying here.
Query: Is this true for all customers? Can you point to any documentation that indicates that base url and token url are expected to be in this format?

Answer: Yes, it true for all the customers and is mentioned on the developer portal itself. Pasting the link but it might require credential to log in to the below link.
Link: https://developer.au.cloud.ariba.com/api/apis